gmail-to-salesforce-lead/main.bal

```diff
@@ -0,0 +1,193 @@
+//Re-order the imports alphabetically
+import ballerina/lang.runtime;
+import ballerina/log;
+import ballerina/mime;
+import ballerinax/googleapis.gmail as gmail;
+import ballerinax/openai.chat as openAI;
+import ballerinax/salesforce as sfdc;
+
+type Email record {|
+    string 'from;
+    string subject;
+    string body;
+|};
+
+type Name record {|
+    string firstName__c;
+    string lastName__c;
+|};
+
+type Lead record {|
+    *Name;
+    string email__c;
+    string phoneNumber__c;
+    string company__c;
+    string designation__c;
+|};
+
+configurable string gmailAccessToken = ?;
+configurable string openAIKey = ?;
+configurable string salesforceBaseUrl = ?;
+configurable string salesforceAccessToken = ?;
+
+final string label = "Lead";
+
+public function main() returns error? {
+    while true {
+        Email[] emails = check getEmails(label);
+
+        Lead[] leads = [];
+        from Email email in emails
+        do {
+            Lead|error lead = generateLead(email.'from, email.subject, email.body);
+            if lead is Lead {
+                leads.push(lead);
+            }
+        };
+
+        check addLeadsToSalesforce(leads);
+
+        runtime:sleep(600);
+    }
+}
+
+function getEmails(string label) returns Email[]|error {
+    gmail:Client gmailClient = check new ({auth: {token: gmailAccessToken}});
```

Comment:
It seems email client is created during each iteration of the loop. instead can we defined a final client outside the main?

Reply:
Fixed in [cce2eaf](https://github.com/ballerina-guides/integration-samples/pull/32/commits/cce2eafff5580b9fc4de69c431a8951e6c90b36a)